corehq/ex-submodules/couchforms/models.py

```diff
@@ -346,27 +346,56 @@ def _meta_to_json(meta):
     def xml_md5(self):
         return hashlib.md5(self.get_xml().encode('utf-8')).hexdigest()
 
-    def archive(self, user_id=None):
-        if self.is_archived:
-            return
-        self.doc_type = "XFormArchived"
-        self.history.append(XFormOperation(
-            user=user_id,
-            operation='archive',
-        ))
-        self.save()
-        xform_archived.send(sender="couchforms", xform=self)
+    def archive(self, user_id=None, retry_archive=False):
+        if not retry_archive:
+            if self.is_archived:
+                return
+            # If this archive was initiated by a user, delete all other stubs for this action so that this action
+            # isn't overridden
+            from couchforms.models import UnfinishedArchiveStub
+            UnfinishedArchiveStub.objects.filter(user_id=user_id).all().delete()
```

Comment:
I think you only need to delete the ones for this form right?

Reply:
Replaced user_id with form_id here: https://github.com/dimagi/commcare-hq/pull/22470/commits/9a3857e686b4c3d205e61e1c3b12a2e77bd09a1e